nncf/quantization/algorithms/weight_compression/algorithm.py

```diff
@@ -440,21 +445,24 @@ def apply(
                 filter(lambda node: node.node_name not in nodes_names_to_exclude, nodes_to_compress)
             )
 
-        if self._awq and activations is not None and self._mode != CompressWeightsMode.E2M1:
+        if self._awq:
             awq_params = self._advanced_parameters.awq_params
             awq_algo = AWQ(
                 model,
                 self._backend_entity.name_to_node_mapping,
                 all_weight_params,
                 nodes_to_compress,
-                activations,
+                self._statistics,
                 awq_params.subset_size,
                 awq_params.percent_to_apply,
                 awq_params.alpha_min,
                 awq_params.alpha_max,
                 awq_params.steps,
             )
             awq_algo.apply(model, graph)
+            # After applying AWQ we need to update statistics since AWQ alters the activations
+            awq_algo.update_statistics(self._statistics)
+            del awq_algo
```

Comment:
I guess it will be happened without explicit deletion on this line.

Reply:
Yes, but without explicit deletion it can be garbage collected only after exiting the `apply()` method. With this, it can be GC'ed right away, before/during GTPQ/SE/LoRA are applied.

Reply:
IMHO: the separate function to run AWQ is more simple solution.

Reply:
I suggest to do refactoring in a separate PR.

Regarding effectiveness of `del` statements. I've added a couple more of them and run an experiment on tiny-llama with AWQ + GPTQ  enabled. Without del's the peak memory is 9884 MiB, with -- 9471 MiB. The difference should roughly equal the size of the statistics collected.